### PR TITLE
adds error for list name exists

### PR DESCRIPTION
**Pull request by HiteshRepo**
adds error for list name exists

#### Does this PR need a docs update or release note?
- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)
#4754 

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/api/graph/errors.go [hunk 1]
```diff
@@ -241,6 +243,11 @@ func IsErrItemAlreadyExistsConflict(err error) bool {
 		parseODataErr(err).hasErrorCode(err, nameAlreadyExists)
 }
 
+func IsErrNameAlreadyExistsConflict(err error) bool {
```
Comment (pandeyabs) [inline src/pkg/services/m365/api/graph/errors.go:246]: Can we just use `IsErrItemAlreadyExistsConflict`? Why do we need this new check? Also, I don't see `ErrNameAlreadyExistsConflict` sentinel being used anywhere. Are you planning on using it in a future PR?

Comment (HiteshRepo) [reply src/pkg/services/m365/api/graph/errors.go:246]: yes @pandeyabs, will use this for `In-place` restore when there is a `list name` collision

Comment (pandeyabs) [reply src/pkg/services/m365/api/graph/errors.go:246]: Thanks! Why not use existing `IsErrItemAlreadyExistsConflict` function? It's also checking against `nameAlreadyExists` odata error. You can reuse the ` graph.ErrItemAlreadyExistsConflict` sentinel as well like the restore code for other services. Let me know if you disagree.

Comment (HiteshRepo) [reply src/pkg/services/m365/api/graph/errors.go:246]: yeah I can use `IsErrItemAlreadyExistsConflict` itself. Closing this PR

**Pull request closed without merging**